File: client/react/hrms/src/services/candidateLanguageService.js
import axios from "axios";

export default class CandidateLanguageService {
  getAll() {
    return axios.get("http://localhost:8080/api/candidatelanguages/getall");
  }

  getAllByCandidateId(candidateId) {
    return axios.get(
      "http://localhost:8080/api/candidatelanguages/getallbycandidateid?candidateId=" +
        candidateId
    );
  }

  save(candidateLink) {
    return axios.post(
      "http://localhost:8080/api/candidatelanguages/save",
      candidateLink
    );
  }

  delete(candidateLink) {
    return axios.post(
      "http://localhost:8080/api/candidatelanguages/delete",
      candidateLink
    );
  }
}

File: server/java/src/main/java/com/hrms/business/abstracts/CandidateLanguageService.java
package com.hrms.business.abstracts;

import com.hrms.core.utilities.results.DataResult;
import com.hrms.core.utilities.results.Result;
import com.hrms.entities.concretes.CandidateLanguage;

import java.util.List;

public interface CandidateLanguageService {
    Result save(CandidateLanguage candidateLanguage);
    Result delete(CandidateLanguage candidateLanguage);
    DataResult<List<CandidateLanguage>> getAll();
    DataResult<List<CandidateLanguage>> getAllByCandidateId(int candidateId);
}
